cpp/src/connect/connectivity.cu

```diff
@@ -0,0 +1,152 @@
+#include "weak_cc.cuh"
+
+#include "utilities/graph_utils.cuh"
+#include "utilities/error_utils.h"
+#include <cugraph.h>
+#include <algo_types.h>
+
+#include <iostream>
+#include <type_traits>
+#include <cstdint>
+
+//
+/**
+ * @brief Compute connected components. 
+ * The weak version was imported from cuML.
+ * This implementation comes from [1] and solves component labeling problem in
+ * parallel on CSR-indexes based upon the vertex degree and adjacency graph.
+ *
+ * [1] Hawick, K.A et al, 2010. "Parallel graph component labelling with GPUs and CUDA"
+ *
+ * @tparam Type the numeric type of non-floating point elements
```

Comment:
Type => IndexT?

Reply:
Please specify file:line and, in general, state more clearly your question, please.

Reply:
line 21, the line right above this comment.

Reply:
I am not sure how you're viewing review comments, but in the web interface, I can click the specific line in the specific file to add the comment for the line. You should be able to connect the comment with the specific line if you are seeing this through the web.

Reply:
@seunghwak , I don't know what operator `=>` means for you. I suspect it means you want me to replace the left side of it with the right side of it. But I had to guess. Wouldn't be more easy if you specified in plain English what you meant?

Reply:
```
line 21 @tparam Type the numeric type of non-floating point elements
line 22 @tparam TPB_X the threads to use per block when configuring the kernel
```

So here, I am guessing that you're giving explanation about template parameters in

```
line 28 template<typename IndexT,
line 29         int TPB_X = 32>
```

So line 21 explains what typename IndexT is and line 22 explains what int TPB_X = 32 is.

So what appears after @tparam in line 21 should be IndexT, not Type. So I am asking whether line 21 should be @tparam IndexT ... instead of @tparam Type ...

Is this clear? Let me know if you need further clarification.

Reply:
Yes, fixed that, just not checked in, yet.